packages/app-config-writer/src/preview-config/ui5-yaml.ts

```diff
@@ -0,0 +1,304 @@
+    //check path and respect defaults
+    if (pathIsFlpPath(path, configuration) && !path?.includes(DEFAULT_FLP_PATH)) {
+        configuration.flp.path = path;
+        writeConfig = true;
+    }
+    //check intent and respect defaults
+    if (intent && `${intent?.object}-${intent?.action}` !== defaultIntent) {
+        configuration.flp.intent = {
+            object: intent.object,
+            action: intent.action
+        };
+        writeConfig = true;
+    }
+
+    if (writeConfig) {
+        newMiddlewareConfig.configuration = configuration;
+    }
+
+    return newMiddlewareConfig as CustomMiddleware<PreviewConfig>;
+}
+
+/**
+ * Updates the preview middleware configurations according to the scripts they are being used in package.json.
+ *
+ * It will process all given UI5 configuration yaml files and check if the preview middleware configuration must be updated based on a given script.
+ * If a script is valid, the preview middleware configuration will be updated and the corresponding file renamed.
+ * UI5 configuration yaml files which are not used in any script will be sanitized.
+ *
+ * @param fs - file system reference
+ * @param basePath - base path to be used for the conversion
+ * @param logger logger to report info to the user
+ */
+export async function updatePreviewMiddlewareConfigs(
+    fs: Editor,
+    basePath: string,
+    logger?: ToolsLogger
+): Promise<void> {
+    const ui5YamlFileNames = await getAllUi5YamlFileNames(basePath, fs);
+    const unprocessedUi5YamlFileNames = [...ui5YamlFileNames];
+    const packageJsonPath = join(basePath, 'package.json');
+    const packageJson = fs.readJSON(packageJsonPath) as Package | undefined;
+    for (const [scriptName, script] of Object.entries(packageJson?.scripts ?? {})) {
+        if (!script || !isValidPreviewScript(scriptName, script)) {
+            continue;
+        }
+
+        const ui5Yaml = basename(extractYamlConfigFileName(script));
+        unprocessedUi5YamlFileNames.splice(unprocessedUi5YamlFileNames.indexOf(ui5Yaml), 1);
+
+        if (
+            !isUi5YamlToBeConverted(ui5Yaml, scriptName, ui5YamlFileNames, logger) ||
+            (await isUi5YamlAlreadyConverted(fs, basePath, ui5Yaml, scriptName, script, logger))
+        ) {
+            continue;
+        }
+
+        try {
+            await processUi5YamlConfig(fs, basePath, ui5Yaml, script);
+        } catch (error) {
+            logger?.warn(
+                `Skipping script '${scriptName}' which refers to UI5 yaml configuration file '${ui5Yaml}'. ${error.message}`
+            );
+            continue;
+        }
+
+        const { path } = extractUrlDetails(script);
+        if (path) {
+            await renameSandbox(fs, join(await getWebappPath(basePath), path), logger);
+        }
+        ensurePreviewMiddlewareDependency(packageJson, fs, packageJsonPath);
+
+        logger?.info(`UI5 yaml configuration file '${ui5Yaml}' updated according to script '${scriptName}'.`);
```

Comment:
```suggestion
        logger?.info(`The UI5 YAML configuration file, '${ui5Yaml}', has been updated according to script, '${scriptName}'.`);
```

Reply:
Done in fe3e4ffa7a57333062a7629ce48b4f040708751f